front/lib/resources/data_source_resource.ts

```diff
@@ -33,31 +34,40 @@ export type FetchDataSourceOptions = {
 export interface DataSourceResource
   extends ReadonlyAttributesType<DataSource> {}
 // eslint-disable-next-line @typescript-eslint/no-unsafe-declaration-merging
-export class DataSourceResource extends BaseResource<DataSource> {
+export class DataSourceResource extends ResourceWithVault<DataSource> {
   static model: ModelStatic<DataSource> = DataSource;
 
-  editedByUser: Attributes<User> | undefined;
+  readonly editedByUser: Attributes<User> | undefined;
 
   constructor(
     model: ModelStatic<DataSource>,
     blob: Attributes<DataSource>,
-    editedByUser?: Attributes<User>
+    vault: VaultResource,
+    { editedByUser }: { editedByUser?: Attributes<User> } = {}
   ) {
-    super(DataSourceResource.model, blob);
+    super(DataSourceResource.model, blob, vault);
+
     this.editedByUser = editedByUser;
   }
 
-  static async makeNew(blob: CreationAttributes<DataSource>) {
+  static async makeNew(
+    blob: Omit<CreationAttributes<DataSource>, "vaultId">,
+    vault: VaultResource
```

Comment:
I would have added the vault in blob and destructured it inside the body, to have one single "blob" parameter containing all the attributes (so, in the end, with vault instead of vaultId).. ?